src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -553,20 +509,16 @@ static void handle_parallel_super_load(Symbol* name,
 }
 
 // Bootstrap and non-parallel capable class loaders use the LOAD_INSTANCE placeholder to
-// wait for parallel class loading and to check for circularity error for Xcomp when loading
-// signature classes.
-// parallelCapable class loaders do NOT wait for parallel loads to complete
+// wait for parallel class loading and/or to check for circularity error for Xcomp when loading.
 static bool needs_load_placeholder(Handle class_loader) {
   return class_loader.is_null() || !is_parallelCapable(class_loader);
 }
 
-// For bootstrap and non-parallelCapable class loaders, check and wait for
-// another thread to complete loading this class.
```

Comment:
A replacement comment describing the method would be nice.

Reply:
Ok, how about
// Check for other threads loading this class either to throw CCE or wait in the case of the boot loader.